lib/protocol/bufferpool.go

```diff
@@ -4,27 +4,67 @@ package protocol
 
 import "sync"
 
+// Global pool to get buffers from. Requires Blocksizes to be initialised,
+// therefore it is initialized in the same init() as BlockSizes
+var thePool bufferPool
+
+// GetBuf returns a byte slice with the requested size from the global pool
+func GetBuf(size int) []byte {
```

Comment:
I'd have a `var protocol.BufferPool = &bufferPool{}` that implements some interface, and use protocol.BufferPool.GetBuffer() etc.

Also, we could just have a separate pool in the model package and separate pool in protocol package (moving the pool implementation to utils or something like that), so that we don't have this ugly reaching into the protocol package. (Or move the cruft into its own package?)

Reply:
I was thinking about the scope of pools, and came up with no good reason to restrict it: The more one pool is used, the higher the chances to save some allocations.

I originally implemented it in the util package, but then I can't base the bucketing directly on block sizes. I can of course redefine them there.

Reply:
@calmh ?